compiler/passes/createTaskFunctions.cpp

```diff
@@ -654,6 +652,28 @@ static void replaceVarUses(Expr* topAst, SymbolMap& vars) {
   }
 }
 
+/********************** helpers **********************/
+
+// Is this expr within a try or try! statement or a 'throws' function?
+static bool inTryContext(Expr* expr) {
+  if (FnSymbol* parentFn = toFnSymbol(expr->parentSymbol)) {
+    if (parentFn->throwsError())
+      return true;
+    if (isTaskFun(parentFn) && ! parentFn->hasFlag(FLAG_OUTSIDE_TRY))
+      return true;
+  }
+
+  Expr* curr = expr->parentExpr; // must be inTree()
```

Comment:
Could we just `INT_ASSERT` that?

Reply:
I suggest it is not worth it. Much of the compiler code relies on this condition without INT_ASSERT-ing it; if the condition does not hold, the compiler would stop rather than silently doing something incorrectly. This case is not different. So by analogy I should probably remove the comment, as the rest of the compiler typically does not have it. :)